### PR TITLE
Fix add_torrent api

### PR DESCRIPTION
**Pull request by wind-mask**
change the add_torrent method and AddTorrentArg and it's source to support torrent from file.

The add_torrent method no longer uses the generic post and request methods, but given that there is only one place in the qbittorrent api that sends raw data, such a hack is acceptable.
Probably clone torrent file bytes unnecessarily. unoptimized considering the size of the torrent file.

Added the test and incidentally fixed the failure of the torrent link in the test.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -490,7 +490,78 @@ impl Qbit {
...
+                self.post("torrents/add", Some(arg.borrow())).await?.end()
+            }
+            TorrentSource::TorrentFiles { torrents } => {
+                {
+                    for i in 0..3 {
+                        // If it's not the first attempt, we need to re-login
+                        self.login(i != 0).await?;
+                        let mut form = reqwest::multipart::Form::new();
+                        // Serialize `a` to a HashMap and add it to the form
+                        let a_map: HashMap<String, String> = {
```
Comment (George-Miao) [inline src/lib.rs]: I prefer handwriting this part, and not all fields are string

File: src/model/torrent.rs [hunk 2]
```diff
@@ -391,7 +391,7 @@ pub enum TorrentSource {
     /// URLs
     Urls { urls: Sep<Url, '\n'> },
     /// Raw data of torrent file.
-    TorrentFiles { torrents: Vec<u8> },
+    TorrentFiles { torrents: Vec<(String, Vec<u8>)> },
```
Comment (George-Miao) [inline src/model/torrent.rs]: Maybe we should make a dedicated struct for this. E.g. 

```
struct Torrent {
  filename: String,
  bytes: Vec<u8>
}
```

Comment (wind-mask) [reply src/model/torrent.rs]: Yes, I will add a
```Rust 
pub struct TorrentFile {
    pub filename: String,
    pub data: Vec<u8>
}
``` 
This helps to understand the parameters.

File: src/lib.rs (hunk 1 above)
Comment (wind-mask) [reply src/lib.rs]: Considering that the `Form::text` method actually accepts `str`
```Rust
reqwest::multipart::Form::text<T, U>(self, name: T, value: U) -> Form
    where
        T: Into<Cow<'static, str>>,
        U: Into<Cow<'static, str>>,
```
Maybe we can be lazy with `serde_json`.I think it's the same as `req.form(body)`(except it's `serde_urlencoded` there).
It shouldn't be possible for the serialization to go wrong here, so I removed the extra expect.
```Rust
let a_map: HashMap<String, Value> = serde_json::from_value(serde_json::to_value(&a)?)?;
    for (k, v) in a_map {
        form = form.text(k, v.to_string());
}
```

Comment (George-Miao) [reply src/lib.rs]: What about keep the as_object, but use for_each and to_string to push the options?

Comment (wind-mask) [reply src/lib.rs]: there's some ownership troubles here.`Form::text`require the ownership of Form.

Comment (George-Miao) [reply src/lib.rs]: `form = form.text(...)` then

Comment (wind-mask) [reply src/lib.rs]: Like this:
```rust
let form = torrents.iter().fold(
    serde_json::to_value(a)?
        .as_object()
        .unwrap()
        .into_iter()
        .fold(reqwest::multipart::Form::new(), |form, (k, v)| {
            form.text(k.to_owned(), v.to_string())
        }),
    |mut form, torrent| {
        let p = reqwest::multipart::Part::bytes(torrent.data.clone())
            .file_name(torrent.filename.clone())
            .mime_str("application/x-bittorrent")
            .unwrap();
        form = form.part("torrents", p);
        form
    },
);
```

Comment (George-Miao) [reply src/lib.rs]: This looks good to me

**Review by George-Miao — APPROVED** [review]
LGTM

**Pull request merged**